cmd/oci/main.go

```diff
@@ -50,5 +60,48 @@ func main() {
 	log.SetOutput(f)
 
 	log.Printf("OCI FlexVolume Driver version: %s (%s)", version, build)
-	flexvolume.ExecDriver(&driver.OCIFlexvolumeDriver{}, os.Args)
+
+	registerDrivers()
+
+	driver, err := getDriverFromArgs()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	flexvolume.ExitWithResult(flexvolume.ExecDriver(driver, os.Args))
+}
+
+func getDriverFromArgs() (flexvolume.Driver, error) {
+	driver, err := getDriver(flexvolume.DefaultSymlinkDirectory) //Block volume is default
+	if err != nil {
+		return nil, err
+	}
+
+	dir := path.Base(os.Args[0])
+	dir = string(strings.TrimPrefix(dir, "oracle~"))
+
+	if dir != "oci" && dir != flexvolume.DefaultSymlinkDirectory {
+		driver, err = getDriver(dir)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	log.Printf("Using %s driver", dir)
+
+	return driver, nil
+}
+
+func registerDriver(name string, driver flexvolume.Driver) {
+	if drivers[name] == nil {
+		drivers[name] = driver
+	}
+}
+
+func getDriver(name string) (flexvolume.Driver, error) {
+	driver := drivers[name]
+	if driver == nil {
```

Comment:
AFAIK this will only be true if `drivers["foo"]` is explicitly assigned to `nil` because nil is not the zero value for `flexvolume.Driver` (and drivers is `map[string]flexvolume.Driver`). Hence the need for checking `ok`.